app/assets/javascripts/short-form/ShortFormNavigationService.js.coffee

```diff
@@ -1,61 +1,147 @@
-    'alternate-contact-name': {path: 'alternate-contact-phone-address'}
-    'alternate-contact-phone-address': {callback: ['goToLandingPage'], params: 'Household'}
+    'prerequisites':
+      callbacks: [
+        Service.goToAndTrackFormSuccess.bind(null, 'dahlia.short-form-application.name')
+      ]
+    'name':
+      scopedCallbacks: [{func: 'checkAfterNamePage'}]
+    'contact':
+      scopedCallbacks: [
+        {func: 'checkIfAddressVerificationNeeded'}
+        {func: 'checkPreferenceEligibility'}
+      ]
+    'verify-address':
+      scopedCallbacks: [{func: 'checkPreferenceEligibility'}]
+      path: 'alternate-contact-type'
+    'alternate-contact-type':
+      scopedCallbacks: [{func: 'checkIfAlternateContactInfoNeeded'}]
+    'alternate-contact-name':
+      path: 'alternate-contact-phone-address'
+    'alternate-contact-phone-address':
+      callbacks: [Service.goToLandingPage.bind(null, 'Household')]
     # household
-    'household-intro': {callback: ['validateHouseholdEligibility'], params: 'householdMatch'}
-    'household-members': {callback: ['validateHouseholdEligibility'], params: 'householdMatch'}
-    'household-member-form': {callback: ['addHouseholdMember', 'checkPreferenceEligibility']}
-    'household-member-form-edit': {callback: ['addHouseholdMember', 'checkPreferenceEligibility']}
-    'household-member-verify-address': {path: 'household-members', callback: ['checkPreferenceEligibility']}
-    'household-public-housing': {callback: ['checkIfPublicHousing']}
-    'household-monthly-rent': {callback: ['checkIfReservedUnits']}
-    'household-reserved-units-veteran': {callback: ['checkIfReservedUnits'], params: RESERVED_TYPES.DISABLED}
-    'household-reserved-units-disabled': {path: 'household-priorities'}
+    'household-intro':
+      scopedCallbacks: [{
+        func: 'validateHouseholdEligibility'
+        param: 'householdMatch'
+      }]
+    'household-members':
+      scopedCallbacks: [{
+        func: 'validateHouseholdEligibility'
+        param: 'householdMatch'
+      }]
+    'household-member-form':
+      scopedCallbacks: [
+        {func: 'addHouseholdMember'}
+        {func: 'checkPreferenceEligibility'}
+      ]
+    'household-member-form-edit':
+      scopedCallbacks: [
+        {func: 'addHouseholdMember'}
+        {func: 'checkPreferenceEligibility'}
+      ]
+    'household-member-verify-address':
+      scopedCallbacks: [{func: 'checkPreferenceEligibility'}]
+      path: 'household-members'
+    'household-public-housing': {scopedCallbacks: [{func: 'checkIfPublicHousing'}]}
+    'household-monthly-rent': {scopedCallbacks: [{func: 'checkIfReservedUnits'}]}
+    'household-reserved-units-veteran':
+      scopedCallbacks: [{
+        func: 'checkIfReservedUnits'
+        param: RESERVED_TYPES.DISABLED
+      }]
+    'household-reserved-units-disabled':
+      path: Service.getPostReservedPage(ShortFormApplicationService.listing)
```

Comment:
I think I might need another pairing session to understand the navigation service. In this case, I think it might actually be clearer to put in a ternary statement rather than calling this function, but I am open to feedback!

I also think I don't fully understand what the difference between callbacks and paths are wonder if it'd be clearer to call these "nextPage" instead of "path".

Reply:
@akegan and i discussed this over slack call and decided that this is as far as we have time to take the navigation refactor for now. further refactor will be done in the future as we continue to work on the short form.